lib/dao.js

```diff
@@ -1413,11 +1413,17 @@ DataAccessObject.removeById = DataAccessObject.destroyById = DataAccessObject.de
 
   var Model = this;
 
-  this.remove(byIdQuery(this, id).where, options, function(err) {
+  this.remove(byIdQuery(this, id).where, options, function(err, data) {
+    var deleted = data && data.count > 0;
     if ('function' === typeof cb) {
-      cb(err);
+      if (options.reportFailureAsError && !err && !deleted) {
+        err = new Error('No instance with id ' + id + ' found for ' + Model.modelName);
+        cb(err, false);
+      } else {
+        cb(err, deleted);
```

Comment:
How about passing directly the `data` object returned by the connector, in the same way as we already do elsewhere?